pkg/localstore/pin.go

```diff
@@ -37,36 +32,34 @@ func (db *DB) PinnedChunks(ctx context.Context, cursor swarm.Address) (pinnedChu
 
 	// send empty response if there is nothing pinned
 	if c == 0 {
-		return pinnedChunks, nil
+		return nil, nil
 	}
 
-	it, err := db.pinIndex.First(prefix)
-	if err != nil {
-		return nil, fmt.Errorf("get first pin: %w", err)
-	}
 	err = db.pinIndex.Iterate(func(item shed.Item) (stop bool, err error) {
-		pinnedChunks = append(pinnedChunks,
+		if offset > 0 {
+			offset--
+			return false, nil
+		}
+		chunks = append(chunks,
 			&storage.Pinner{
 				Address:    swarm.NewAddress(item.Address),
 				PinCounter: item.PinCounter,
 			})
-		count++
-		if count >= maxChunksToDisplay {
+		limit--
+
+		if limit == 0 {
 			return true, nil
-		} else {
-			return false, nil
 		}
-
+		return false, nil
 	}, &shed.IterateOptions{
-		StartFrom:         &it,
 		SkipStartFromItem: false,
```

Comment:
No need for `SkipStartFromItem` option if the `StartFrom` is removed.
Options here can be replaced by `nil`.